File: leet_594_longest_subseq___Map.ts

function findLHS(nums: number[]): number {
    let mp = new Map<number, number>()
    for (let n of nums) {
        if (n in mp)
            mp[n]++
        else
            mp[n] = 1
    }
    // console.log(mp)
    let res = 0
    for (let n of nums) {
        if (n in mp && n + 1 in mp)
            res = Math.max(res, mp[n] + mp[n + 1])
    }
    return res
};

File: leet_594_longest_subseq___map.cc
class Solution {
public:
    int findLHS(vector<int>& nums)
    {
        map<int, int>   mp;
        int             res;

        res = 0;
        for (int n: nums)
            mp[n]++;
        for (int n: nums)
        {
            if (mp.find(n + 1) != mp.end())
                res = max(res, mp[n] + mp[n + 1]);
        }
        return res;
    }
};
